modules/social.py

```diff
@@ -36,13 +36,13 @@ def __init__(self, bot):
         self.inprogressEdits = {}
         self.letterCodepoints = ['1f1e6', '1f1e7', '1f1e8', '1f1e9', '1f1ea', '1f1eb', '1f1ec', '1f1ed', '1f1ee', '1f1ef', '1f1f0', '1f1f1', '1f1f2', '1f1f3', '1f1f4', '1f1f5', '1f1f6', '1f1f7', '1f1f8', '1f1f9', '1f1fa', '1f1fb', '1f1fc', '1f1fd', '1f1fe', '1f1ff']
         
-        # Friendcode regex for profile editor (lenient) and chat filter (matches separators between digits), \u2014 = em-dash
-        self.friendcodeProfileRe = re.compile(r'(?:sw)?[ \-\u2014]?(\d{4})[ \-\u2014]?(\d{4})[ \-\u2014]?(\d{4})', re.I)
-        self.friendcodeFilterRe = re.compile(r'(?:sw)?[ \-\u2014]?(\d{4})[ \-\u2014](\d{4})[ \-\u2014](\d{4})', re.I)
-
-        # Friendcode regex for profile editor (lenient) and chat filter (matches separators between digits), \u2014 = em-dash
-        self.friendcodeProfileRe = re.compile(r'(?:sw)?[ \-\u2014]?(\d{4})[ \-\u2014]?(\d{4})[ \-\u2014]?(\d{4})', re.I)
-        self.friendcodeFilterRe = re.compile(r'(?:sw)?[ \-\u2014]?(\d{4})[ \-\u2014](\d{4})[ \-\u2014](\d{4})', re.I)
+        self.friendCodeRegex = { # Friend Code Regexs (\u2014 = em-dash)
+            # Profile setup/editor (lenient)
+            "profile": re.compile(r'(?:sw)?[ \-\u2014]?(\d{4})[ \-\u2014]?(\d{4})[ \-\u2014]?(\d{4})', re.I),
+            # Chat filter, "It appears you've sent a friend code." Requires separators and discards AC designer prefixes.
+            # Discards Animal Crossing's MA & MO designer codes. Thanks to Jade for pointing out alternation and negated sets.
+            "chatFilter": re.compile(r'(?:sw|m[^ao]|[^M][\w]|^\w|^)[ \-\u2014]?(\d{4})[ \-\u2014](\d{4})[ \-\u2014](\d{4})', re.I + re.M)
```

Comment:
Appears to fail in some cases. Matches "ma1234 1234 1234". I have not exhaustively tested different combinations but it might be a good idea to test every combination.

Reply:
Doesn't match "ma1234 1234 1234" for me, have you made sure you have the case insensitive flag set?

Reply:
nope.